### PR TITLE
eslint-changed: Make it a project

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

Finally got around to making this a project (and writing tests for it).

Not figuring out how to push this to the NPM repo yet, but this is a step along that path.

#### Jetpack product discussion

https://github.com/Automattic/jetpack/pull/20085#issuecomment-861928293

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Try running it, does it work?
* Does the package name "@automattic/eslint-changed" for it make sense?

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
This looks good to me. I've added a few comments, nitpicks really. I figure the readme will be an important entry point for folks interested in the package, so we may as well polish it as much as we can :)

> Does the package name "@automattic/eslint-changed" for it make sense?

I think so, yes. Sounds perfect to me, it does what it says on the tin, and will not cause confusions with other tools (e.g. [`lint-diff`](https://github.com/grvcoelho/lint-diff) or [`lint-staged`](https://github.com/okonet/lint-staged)).

I was wondering if we should add some examples in the usage section, just like those 2 other packages do? What do you think?

File: projects/js-packages/eslint-changed/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+set environment variables `GIT` and/or `ESLINT` as appropriate.
+
+The following options are used with manual mode:
+
+* `--git`: Signify that you're using git mode.
+* `--git-staged`: Compare the staged version to the HEAD version (this is the default).
+* `--git-unstaged`: Compare the working copy version to the staged (or HEAD) version.
+* `--git-base <ref>`: Compare the HEAD version to the HEAD of a different base (e.g. branch).
+
+## Inspiration
```
[the remark this answers is not in the document]
Comment (anomiex) [reply projects/js-packages/eslint-changed/README.md:41]: I have no idea, do we?

  github-actions/repo-gardening, plugins/backup, and plugins/beta have a "License" section. None of the rest of the monorepo projects do. Unless we're intending on changing that, I'm inclined to just let the LICENSE.txt handle it.

File: projects/js-packages/eslint-changed/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Eslint-changed
```
[the remark this answers is not in the document]
Comment (anomiex) [reply projects/js-packages/eslint-changed/README.md]: Sure

File: projects/js-packages/eslint-changed/bin/eslint-changed.js [hunk 3]
```diff
@@ -213,6 +215,10 @@ async function main() {
 		diff = parseDiff( fs.readFileSync( argv.diff, 'utf8' ) );
 		diffBase = argv.diffBase || process.cwd();
 		files = getFilesFromDiff( diff );
+		if ( program.args.length ) {
+			const cmdLineFiles = new Set( program.args );
+			files = files.filter( file => cmdLineFiles.has( file ) );
+		}
 		debug( 'Determined files from diff:', files );
```
[the remark this answers is not in the document]
Comment (anomiex) [reply projects/js-packages/eslint-changed/bin/eslint-changed.js:222]: Sure. Change incoming.

File: projects/js-packages/eslint-changed/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeherve) [reply projects/js-packages/eslint-changed/README.md:81]: You're right, there doesn't seem to be a hard rule for that. Let's do without it for the moment; in any case the license is already mentioned in the `package.json` file as well as the license file itself. 👍

**Review by jeherve — APPROVED** [review]

**Pull request merged**